src/bin/pg_dump/dump_babel_utils.c

```diff
@@ -993,47 +974,70 @@ prepareForBabelfishDatabaseDump(Archive *fout, SimpleStringList *schema_include_
 void
 setBabelfishDependenciesForLogicalDatabaseDump(Archive *fout)
 {
+	int        		i;
 	PQExpBuffer		query;
 	PGresult		*res;
-	TableInfo		*extprop_table;
-	TableInfo		*sysdb_table;
-	TableInfo		*namespace_ext_table;
-	TableInfo		*schema_perms_table;
-	DumpableObject		*dobj;
-	DumpableObject		*refdobj;
+	TableInfo		*sysdb_table = NULL;
+	DumpableObject		*refdobj = NULL;
 
 	if (!isBabelfishDatabase(fout) || fout->dopt->binary_upgrade || fout->dopt->dataOnly)
 		return;
 
 	query = createPQExpBuffer();
-	/* get oids of sys.babelfish_sysdatabases, sys.babelfish_namespace_ext, babelfish_extended_properties, babelfish_schema_permissions tables */
+	/*
+	 * Get OIDs for following desired catalog tables:
+	 * sys.babelfish_sysdatabases
+	 * sys.babelfish_namespace_ext
+	 * babelfish_extended_properties
+	 * babelfish_schema_permissions
+	 */
 	appendPQExpBufferStr(query,
```

Comment:
it would be simple to get sysdatabase in a separate call and remaining catalogs in other call. That way later loops are simplified

Reply:
Replied to the Sumit's comment about the why I initially kept it as this. Suggestion taken.